aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -362,33 +426,74 @@ sub pxeprint
     $fh->print (<<EOF
 # File generated by pxelinux AII plug-in.
 # Do not edit.
-default $t->{label}
-    label $t->{label}
-    kernel $t->{kernel}
+default Install $pxe_config->{label}
+    label $pxe_config->{label}
+    kernel $pxe_config->{kernel}
     $appendtxt
 EOF
            );
     # TODO is ksdevice still mandatory? if not, fix schema (code is already ok)
     # ksdecvice=bootif is an anaconda-ism, but can serve general purpose
-    $fh->print ("    ipappend 2\n") if ($t->{ksdevice} && $t->{ksdevice} eq 'bootif');
+    $fh->print ("    ipappend 2\n") if ($pxe_config->{ksdevice} && $pxe_config->{ksdevice} eq 'bootif');
     $fh->close();
 }
 
-# Prints an IP address in hexadecimal.
-sub hexip
+
+# Write the Grub2 configuration file.
+# Return 1 if the file was written successfully, 0 otherwise.
+# TODO: handle append options?
+sub write_grub2_config
 {
-    my $ip = shift || "";
+    my ($self, $cfg) = @_;
+    my $pxe_config = $cfg->getElement (PXEROOT)->getTree;
+
+    my $linux_cmd = $this_app->option(GRUB2_EFI_LINUX_CMD);
+    unless ( $linux_cmd ) {
+        $this_app->error("AII option ".GRUB2_EFI_LINUX_CMD." undefined");
+        return 0;
+    };
+    my $initrd_cmd = $this_app->option(GRUB2_EFI_INITRD_CMD);
+    unless ( $initrd_cmd ) {
+        $this_app->error("AII option ".GRUB2_EFI_INITRD_CMD." undefined");
+        return 0;
+    };
+    my $kernel_root = '';
+    if ( option_exists(GRUB2_EFI_KERNEL_ROOT) ) {
+        $kernel_root = $this_app->option(GRUB2_EFI_KERNEL_ROOT);
+    }
+    my $kernel_path = "$kernel_root/$pxe_config->{kernel}";
+    my $initrd_path = "$kernel_root/$pxe_config->{initrd}";
+
+    my $fh = CAF::FileWriter->open (filepath ($cfg, PXE_VARIANT_GRUB2),
+                                    log => $this_app, mode => 0644);
```

Comment:
https://github.com/quattor/aii/pull/225#discussion_r103656791... I prefer to do it in one go rather than introducing one more inconsistency in this old piece of code!